### PR TITLE
Fix error caused by unhashable error list

### PR DESCRIPTION
**Pull request by sawsa307**
Fix error caused by unhashable error aggregation in getErrorStateReason()

### CODE REVIEW

File: pkg/neg/syncers/transaction.go [hunk 1]
```diff
@@ -347,6 +347,10 @@ func (s *transactionSyncer) ensureNetworkEndpointGroups() error {
 
 // getErrorStateReason returns the reason of the error state based returned error
 func (s *transactionSyncer) getErrorStateReason(err error) string {
+	// these errors only occurs on NEG creation, and they are not related to endpoint sync
```
Comment (swetharepakula) [inline pkg/neg/syncers/transaction.go:350]: we should indicate that they are also not a relevant error state reason.

Comment (sawsa307) [reply pkg/neg/syncers/transaction.go]: Updated. Thanks!

**Review by swetharepakula — COMMENTED** [review]
/lgtm
/approve
/ok-to-test

**Pull request merged**